nbviewer/app.py

```diff
@@ -159,106 +172,138 @@ def init_tornado_application(self):
-            hub_api_token=os.getenv('JUPYTERHUB_API_TOKEN'),
-            hub_api_url=os.getenv('JUPYTERHUB_API_URL'),
-            hub_base_url=os.getenv('JUPYTERHUB_BASE_URL'),
-            ipywidgets_base_url=options.ipywidgets_base_url,
-            jupyter_widgets_html_manager_version=options.jupyter_widgets_html_manager_version,
-            jupyter_js_widgets_version=options.jupyter_js_widgets_version,
-            content_security_policy=options.content_security_policy,
-            binder_base_url=options.binder_base_url,
+                  config=self.config,
+                  index=self.index,
+                  max_cache_uris=self.max_cache_uris,
+                  static_path=self.static_path,
+                  static_url_prefix=self.static_url_prefix,
         )
-    
+        # input computed properties to settings
+        settings.update(
+                  base_url=self.base_url,
+                  cache=self.cache,
+                  client=self.client,
+                  fetch_kwargs=self.fetch_kwargs,
+                  formats=self.formats,
+                  frontpage_setup=self.frontpage_setup,
+                  jinja2_env=self.env,
+                  pool=self.pool,
+                  rate_limiter=self.rate_limiter,
+        )
+        # input settings from CLI options
+        settings.update(
+                  binder_base_url=options.binder_base_url,
+                  cache_expiry_max=options.cache_expiry_max,
+                  cache_expiry_min=options.cache_expiry_min,
+                  content_security_policy=options.content_security_policy,
+                  default_format=options.default_format,
+                  ipywidgets_base_url=options.ipywidgets_base_url,
+                  jupyter_js_widgets_version=options.jupyter_js_widgets_version,
+                  jupyter_widgets_html_manager_version=options.jupyter_widgets_html_manager_version,
+                  localfile_any_user=options.localfile_any_user,
+                  localfile_follow_symlinks=options.localfile_follow_symlinks,
+                  localfile_path=os.path.abspath(options.localfiles),
+                  mathjax_url=options.mathjax_url,
+                  provider_rewrites=options.provider_rewrites,
+                  providers=options.providers,
+                  render_timeout=options.render_timeout,
+                  statsd_host=options.statsd_host,
+                  statsd_port=options.statsd_port,
+                  statsd_prefix=options.statsd_prefix,
+        )
+        # additional settings
+        settings.update(
+                  google_analytics_id=os.getenv('GOOGLE_ANALYTICS_ID'),
+                  gzip=True,
+                  hub_api_token=os.getenv('JUPYTERHUB_API_TOKEN'),
+                  hub_api_url=os.getenv('JUPYTERHUB_API_URL'),
+                  hub_base_url=os.getenv('JUPYTERHUB_BASE_URL'),
+                  log_function=log_request,
+        )
+        log.app_log.info(settings['hub_base_url'])
```

Comment:
If the intention is to keep this log statement, it would be good to prefix it with `JupyterHub Base URL: %s` or some such.

Reply:
Oh oops no that was a goof. Had that there when I was debugging something but forgot to remove it. Sorry about that.

Reply:
Deleted that -- thank you for pointing this out!